frontend/public/components/utils/storage-class-dropdown.tsx

```diff
@@ -0,0 +1,206 @@
+      };
+      const unorderedItems = {};
+      const noStorageClass = 'No storage class';
+      _.each(resources, ({ data }, kindLabel) => {
+        _.reduce(
+          data,
+          (acc, resource) => {
+            acc[resource.metadata.name] = {
+              kindLabel,
+              name: resource.metadata.name,
+              description: resource.metadata.annotations ? resource.metadata.annotations.description : '',
+              default: resource.metadata.annotations ? resource.metadata.annotations['storageclass.kubernetes.io/is-default-class'] === 'true' ? true : false : '',
+              accessMode: resource.metadata.annotations ? resource.metadata.annotations['storage.alpha.openshift.io/access-mode'] : '',
+              provisioner: resource.provisioner,
+              type: resource.parameters ? resource.parameters.type : '',
+              zone: resource.parameters ? resource.parameters.zone : '',
+            };
+            return acc;
+          },
+          unorderedItems
+        );
+      });
+
+      // Determine if there is a default storage class
+      _.each(unorderedItems, (item, name) => {
+        if ((item as any).default) {
+          state.defaultClass = name;
+        }
+      });
+      const { selectedKey } = this.state;
+      if (!state.defaultClass) {
+        // Add No Storage Class option if there is not a default storage class
+        unorderedItems[''] = { kindLabel: '', name: noStorageClass };
+      }
+
+      if (!this.props.loaded || !selectedKey ) {
+        state.title = <span className="text-muted">{nextProps.placeholder}</span>;
+      }
+
+      if (selectedKey) {
+        const item = unorderedItems[selectedKey];
+        if (item) {
+          state.title = this.getTitle(item.kindLabel, item.name);
+        }
+      }
+
+      if (!selectedKey && state.defaultClass) {
+        const item = unorderedItems[state.defaultClass];
+        if (item) {
+          state.title = this.getTitle(item.kindLabel, item.name);
+        }
+      }
+
+      Object.keys(unorderedItems).sort().forEach(function(key) {
+        state.items[key] = unorderedItems[key];
+      });
+      this.setState(state);
+    }
+
+    componentDidUpdate() {
+      const {defaultClass, selectedKey } = this.state;
+      if (!selectedKey && defaultClass) {
+        this.onChange(defaultClass);
+      }
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+      return !_.isEqual(this.state, nextState);
+    }
+
+    autocompleteFilter = (text, item) => fuzzy(text, item.props.name);
+
+    getTitle = (kindLabel, name) => {
+      return kindLabel ? <ResourceName kind="StorageClass" name={name} /> : <span >{name}</span>;
+    }
+
+    onChange = key => {
+      const { name, kindLabel, accessMode } = _.get(this.state, ['items', key], {});
+      this.setState({
+        selectedKey: key,
+        title: this.getTitle(kindLabel, name),
+      });
+      this.props.onChange(key, accessMode);
```

Comment:
Do we still have a reference to the storage class object? If we need more than a name, it would be better to pass the object and let the caller read the access mode if it needs it. (We can create a helper so the annotation logic is in only one place.)

Otherwise you're making this component more tightly coupled to the form. It shouldn't know that the embedding page does anything with access mode.